providers/implementations/ciphers/cipher_aes_xts_hw.c

```diff
@@ -221,9 +223,9 @@ static const PROV_CIPHER_HW aes_xts_rv32i_zbkb_zknd_zkne = {                   \
     cipher_hw_aes_xts_copyctx                                                  \
 };
 # define PROV_CIPHER_HW_select_xts()                                           \
-if (RV32I_ZBKB_ZKND_ZKNE_CAPABLE)                                              \
+if (RISCV_HAVE_ZBKB_AND_ZKND_AND_ZKNE())                                       \
```

Comment:
What benefit do the `()` provide?
We generally use macros without `()` for this kind of thing.

Reply:
They are just a technical necessity. We generate them in the form of inline functions:
```c
# define RISCV_DEFINE_CAP(NAME, INDEX, BIT_INDEX)                   \
    static inline int RISCV_HAVE_##NAME(void)                        \
    {                                                               \
        return (OPENSSL_riscvcap_P[INDEX] & (1 << BIT_INDEX)) != 0; \
    }
# include "riscv_arch.def"
```

In `riscv_arch.def`, we just list the extensions (e.g. `RISCV_DEFINE_CAP(ZKT, 0, 13)`).

As macros cannot be generated like these inline functions, I don't see how we can avoid the `()`.

I agree that some architectures have pure test-statement-macros (e.g. `PMULL_CAPABLE`), but there is also plenty of code that inlines such tests (e.g. `((OPENSSL_ia32cap_P[1] >> 22) & 0x41) == 0x41`). I believe the generated inline functions are a good compromise. And they give a clear indication that these macros are run-time properties and won't be used for compile-time tests (e.g. `if defined (RISCV_HAVE_foo)`) by accident.

If you insist, I will change this to a manually written list of test macros. Just let me know.

Reply:
I don't insist.  Leave it as is, I was wondering why and you've provided a good reason.